app/src/main/java/com/doublesp/coherence/actions/ListFragmentActionHandler.java

```diff
@@ -97,6 +90,9 @@ public void afterTextChanged(Editable s) {
         if (s.subSequence(i - 1, i).toString().equals("\n")) {
             mIdeaInteractor.addIdea(s.toString().trim());
             s.clear();
+            List<Idea> ideas = mIdeaInteractor.getPlan().getIdeas();
+            mShoppingListDatabaseReference.child(mIdeaInteractor.getPlan().getId()).child(
+                    ConstantsAndUtils.IDEAS).setValue(ideas);
```

Comment:
That will not work because in line 122 we are saving a plan, these are individual ideas in the plan. So that will not work. may be i should understand more from you so that i am not missing, the changes here work perfectly fine with no issues, so may be i will understand more before i make that change as you are mentioning.